client/src/main/java/io/pravega/client/batch/impl/BatchClientFactoryImpl.java

```diff
@@ -46,25 +46,22 @@
 
 @Beta
 @Slf4j
-@SuppressWarnings("deprecation")
 public class BatchClientFactoryImpl implements BatchClientFactory {
 
     private final Controller controller;
     private final ConnectionFactory connectionFactory;
     private final SegmentInputStreamFactory inputStreamFactory;
     private final SegmentMetadataClientFactory segmentMetadataClientFactory;
     private final StreamCutHelper streamCutHelper;
-
-    @GuardedBy("this")
-    private final AtomicReference<String> latestDelegationToken;
+    private final AtomicReference<DelegationTokenProvider> delegationTokenProvider;
```

Comment:
The `DelegationTokenProvider` implementation holds the `DelegationToken` under an `AtomicReference`, so using an `AtomicReference` here is redundant and should be removed. 

https://github.com/pravega/pravega/blob/ad9260d5d2e58e7262732818b8b2bf71ef325198/client/src/main/java/io/pravega/client/security/auth/JwtTokenProviderImpl.java#L71

Reply:
Fixed it. Removed the redundant AtomicReference.